src/algebraic_geometry/Scheme.lean

```diff
@@ -124,11 +136,84 @@ lemma Γ_obj_op (X : Scheme) : Γ.obj (op X) = X.presheaf.obj (op ⊤) := rfl
 lemma Γ_map_op {X Y : Scheme} (f : X ⟶ Y) :
   Γ.map f.op = f.1.c.app (op ⊤) := rfl
 
--- PROJECTS:
--- 1. Construct `Spec ≫ Γ ≅ functor.id _`.
--- 2. Adjunction between `Γ` and `Spec`.
---
+section basic_open
+
+variables (X : Scheme) {V U : opens X.carrier} (f g : X.presheaf.obj (op U))
+
+/-- The subset of the underlying space where the given section does not vanish. -/
+def basic_open : opens X.carrier := X.to_LocallyRingedSpace.to_RingedSpace.basic_open f
+
+@[simp]
+lemma mem_basic_open (x : U) : ↑x ∈ X.basic_open f ↔ is_unit (X.presheaf.germ x f) :=
+RingedSpace.mem_basic_open _ _ _
+
+@[simp]
+lemma mem_basic_open_top (f : X.presheaf.obj (op ⊤)) (x : X.carrier) :
+  x ∈ X.basic_open f ↔ is_unit (X.presheaf.germ (⟨x, trivial⟩ : (⊤ : opens _)) f) :=
+RingedSpace.mem_basic_open _ f ⟨x, trivial⟩
+
+@[simp]
+lemma basic_open_res (i : V ⟶ U) :
+  X.basic_open (X.presheaf.map i.op f) = V ∩ X.basic_open f :=
+RingedSpace.basic_open_res _ i.op f
+
+@[simp, priority 1100]
```

Comment:
Should this get a similar comment? Is there some other simp lemma that can be added so that the simp set will be confluent even if `basic_open_res` fires first?

Reply:
I don't think there are such simp lemmas. We will loose the fact that `V = U` after rewriting with `basic_open_res`.

Reply:
```suggestion
-- This should fire before `basic_open_res`.
@[simp, priority 1100]
```

Reply:
Added already :)